lib/elixir/lib/module/locals_tracker.ex

```diff
@@ -122,24 +122,28 @@ defmodule Module.LocalsTracker do
     acc
   end
 
-  defp collect_warnings({tuple, kind, meta, 0}, acc, reachable) do
+  defp collect_warnings({tuple, kind, {meta, _head_meta}, 0}, acc, reachable) do
     if Map.has_key?(reachable, tuple) do
       acc
     else
       [{meta, {:unused_def, tuple, kind}} | acc]
     end
   end
 
-  defp collect_warnings({tuple, kind, meta, default}, acc, reachable) when default > 0 do
+  defp collect_warnings({tuple, kind, {meta, nil}, default}, acc, reachable) do
+    collect_warnings({tuple, kind, {meta, meta}, default}, acc, reachable)
+  end
+
+  defp collect_warnings({tuple, kind, {meta, head_meta}, default}, acc, reachable) do
     {name, arity} = tuple
     min = arity - default
     max = arity
 
     case min_reachable_default(max, min, :none, name, reachable) do
       :none -> [{meta, {:unused_def, tuple, kind}} | acc]
```

Comment:
I guess I could change all to use the head meta, but I'm not sure what is best

Reply:
Let's go with head meta  everywhere and see how it impacts the existing suite and see if the warnings get worse or better.